src/joinboost/executor.py

```diff
@@ -505,61 +518,38 @@ def join(self, intermediates, join_conds, join_type, tables_to_cross, tables_to_
-                    table2].columns) or \
-                        (cond[1].split('.')[1] in intermediates[table1].columns and cond[0].split('.')[1] in
-                         intermediates[table2].columns):
-                    # join the two tables
-                    temp = intermediates[table1].merge(intermediates[table2],
-                                                       how=join_type.lower(), left_on=cond[0].split('.')[1],
-                                                       right_on=cond[1].split('.')[1],
-                                                       suffixes=('', '_drop')).filter(regex='^(?!.*_drop)')
-                    # check if one join condition is qualified and the other is not
-                elif (cond[0] in intermediates[table1].columns and cond[1].split('.')[1] in intermediates[
-                    table2].columns):
-                    # join the two tables
-                    temp = intermediates[table1].merge(intermediates[table2],
-                                                       how=join_type.lower(), left_on=cond[0],
-                                                       right_on=cond[1].split('.')[1],
-                                                       suffixes=('', '_drop')).filter(regex='^(?!.*_drop)')
-                elif (cond[1] in intermediates[table2].columns and cond[0].split('.')[1] in intermediates[
-                    table1].columns):
-                    # join the two tables
-                    temp = intermediates[table1].merge(intermediates[table2],
-                                                       how=join_type.lower(), left_on=cond[1],
-                                                       right_on=cond[0].split('.')[1],
-                                                       suffixes=('', '_drop')).filter(regex='^(?!.*_drop)')
-
-
-                # unqualify all columns in temp. This is to avoid nested columns being qualified with the table name
-                if temp is not None:
-                    for col in temp.columns:
-                        temp = temp.rename(columns={col: col.split('.')[-1]})
-                    # if there are duplicate columns, drop them
-                    temp = temp.loc[:, ~temp.columns.duplicated()]
 
+        # unqualify all join conditions
+        temp_join_conds = [[cond[0].split('.')[-1], cond[1].split('.')[-1]] for cond in join_conds]
+        # flatten temp_join_conds
+        temp_join_conds = [item for sublist in temp_join_conds for item in sublist]
 
+        # sort tables_to_join by the number of times their respective columns appear in join_conds.
+        # TODO: this is a hacky way to avoid duplicate joining between tables.
+        tables_to_join = sorted(tables_to_join,
+                                key=lambda x: len([col for col in intermediates[x].columns if col in temp_join_conds]),
```

Comment:
What is this doing? Is it joining fact table first? Why order by it?

Reply:
Yes, it's ensuring that each merge happens to the fact table. I don't remember the exact reason, but I think it's because previously, there was a chance of applying the same join condition twice. Let me look into this and see if I can remove it.